File: packages/expo-updates/android/src/main/java/expo/modules/updates/selectionpolicy/LauncherSelectionPolicySingleUpdate.java
package expo.modules.updates.selectionpolicy;

import org.json.JSONObject;

import java.util.List;
import java.util.UUID;

import expo.modules.updates.db.entity.UpdateEntity;

/**
 * Trivial LauncherSelectionPolicy that will choose a single predetermined update to launch.
 */
public class LauncherSelectionPolicySingleUpdate implements LauncherSelectionPolicy {

  private UUID mUpdateID;

  public LauncherSelectionPolicySingleUpdate(UUID updateID) {
    mUpdateID = updateID;
  }

  @Override
  public UpdateEntity selectUpdateToLaunch(List<UpdateEntity> updates, JSONObject filters) {
    for (UpdateEntity update : updates) {
      if (update.id.equals(mUpdateID)) {
        return update;
      }
    }
    return null;
  }
}

File: packages/expo-updates/android/src/main/java/expo/modules/updates/selectionpolicy/LauncherSelectionPolicySingleUpdate.kt
package expo.modules.updates.selectionpolicy

import expo.modules.updates.db.entity.UpdateEntity
import org.json.JSONObject
import java.util.*

/**
 * Trivial LauncherSelectionPolicy that will choose a single predetermined update to launch.
 */
class LauncherSelectionPolicySingleUpdate(private val updateID: UUID) : LauncherSelectionPolicy {
  override fun selectUpdateToLaunch(
    updates: List<UpdateEntity>,
    filters: JSONObject?
  ): UpdateEntity? {
    return updates.find { it.id == updateID }
  }
}
